src/resources.rs

```diff
@@ -245,6 +246,14 @@ impl DeviceResources {
     }
 }
 
+impl Deref for DeviceResources {
+    type Target = Vec<Resource>;
```

Comment:
Hmm, any reason to `deref` into a `&Vec<Resource>` and not a slice (`&[Resource]`)?

Reply:
good suggestion, will change to return a slice.